src/components/ProductCard.tsx

```diff
@@ -1,58 +1,107 @@
+import { Trash2, Edit } from "lucide-react";
 import Card from "react-bootstrap/Card";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { notifyDeleteProduct } from "../utils/NotificationUtils";
+import { ProductListProps } from "../types/types";
+import { Button } from "react-bootstrap";
 
-type ProductListProps = {
-  id: string;
-  name: string;
-  business: string[];
-  regions: string[];
-  withLink: boolean;
-  customStyles?: {
-    cardStyle: React.CSSProperties;
-    cardBodyStyle: React.CSSProperties;
-  };
-};
-
-export function ProductList({
+export function ProductCard({
   id,
   name,
   business,
   regions,
-  withLink = true,
-  customStyles,
+  isDelete = true,
+  deleteProduct,
+  status,
+  isAddNewProduct = false,
 }: ProductListProps) {
-  const cardStyle = customStyles?.cardStyle || {};
-  const cardBodyStyle = customStyles?.cardBodyStyle || {};
-  
-  const cardContent = (
-    <Card
-      style={{
-        ...cardStyle,
-        width: "20rem",
-        boxShadow: "0 4px 6px rgba(0, 0, 0, 0.1)",
-        height:"10rem"
-      }}
-    >
-      <Card.Body style={{ ...cardBodyStyle }}>
-        <Card.Title>{name}</Card.Title>
-        <Card.Text>Business: {business.join(", ")}</Card.Text>
-        <Card.Text>Regions: {regions.join(", ")}</Card.Text>
-      </Card.Body>
-    </Card>
-  );
-  if (withLink) {
-    return (
-      <div key={id} style={{ margin: "10px" }}>
-        <Link to={`/product/${id}`} style={{ textDecoration: "none" }}>
-          {cardContent}
-        </Link>
-      </div>
-    );
-  } else {
+  const navigate = useNavigate();
+
+  const handleDelete = (e: React.MouseEvent) => {
+    e.preventDefault();
+    notifyDeleteProduct({ id, name }, () => deleteProduct(id));
+  };
```

Comment:
are we doing prop drilling can we directly use deleteProduct from context?

Reply:
done